app/javascript/lib/search-utils.js

```diff
@@ -0,0 +1,42 @@
+import stringSimilarity from 'string-similarity'
+
+/**
+ * Get list of autocomplete suggestions, based on input text
+ *
+ * Returns top 20 matches: exact prefix matches, then similar matches
+ *
+ * @param {String} inputString String typed by user into text input
+ * @param {Array<String>} targets List of strings to match against
+ */
+export function getAutocompleteSuggestions(inputText, targets) {
+  // Autocomplete when user starts typing
+  if (!targets || !inputText) {
+    return []
+  }
+
+  const text = inputText.toLowerCase()
+
+  const exactMatch = targets.find(gene => gene === inputText)
+
+  // Get genes that start with the input text
+  const prefixMatches =
+    targets.filter(gene => {
+      return gene !== inputText && gene.toLowerCase().startsWith(text)
+    })
+
```

Comment:
I don't know how expensive the simila operations are, but it might be worth skipping the lines below if we already have >20 prefix matches (which I suspect will often be the case when they've just typed one letter)

Reply:
Done, good idea. It's very fast, but why not optimize?